src/lib/io/sol-spi-impl-linux.c

```diff
@@ -152,13 +153,13 @@ sol_spi_transfer(struct sol_spi *spi, const uint8_t *tx, uint8_t *rx, size_t siz
 
 #ifdef WORKER_THREAD
     spi->transfer.worker = sol_worker_thread_new(&config);
-    SOL_NULL_CHECK(spi->transfer.worker, false);
+    SOL_NULL_CHECK(spi->transfer.worker, -ENOMEM);
```

Comment:
Well, we can do that later in another patch. This will involve reviewing all uses of sol_worker_thread_new()